From 8c2e619fe5d7cc9e32f4d34ad93c79cb6b680e8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Thu, 20 Nov 2014 12:58:49 +0100
Subject: [PATCH 36/82] thread: winstore: Do not discard a signaled object.

When a thread is cancelled yet not killable, discarding a WAIT_OBJECT_X would cause the thread to wait forever, since vlc_testcancel would not abort the thread.
---
 src/win32/thread.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/win32/thread.c b/src/win32/thread.c
index 7698675..cab973e 100644
--- a/src/win32/thread.c
+++ b/src/win32/thread.c
@@ -82,7 +82,7 @@ static DWORD vlc_WaitForMultipleObjects (DWORD count, const HANDLE *handles,
             ret = WaitForMultipleObjectsEx (count, handles, FALSE, new_delay, TRUE);
             if (delay != INFINITE)
                 delay -= new_delay;
-            if (isCancelled())
+            if (ret == WAIT_TIMEOUT && isCancelled())
                 ret = WAIT_IO_COMPLETION;
         } while (delay && ret == WAIT_TIMEOUT);
 #else
-- 
2.2.0

